assets/common.sh

```diff
@@ -79,6 +79,26 @@ setup_kubectl() {
     exe kubectl config set-context "$CONTEXT_NAME" --user="$AUTH_NAME" --cluster="$CLUSTER_NAME"
 
     exe kubectl config use-context "$CONTEXT_NAME"
+
+    # Optional. Use the AWS EKS authenticator
+    local use_aws_iam_authenticator
```

Comment:
I don't feel the need `use_aws_iam_authenticator` and `aws_eks_cluster_name` source params. If you'd like to use aws-iam-authenticator, you can specify so in `kubeconfig` source param:

```yaml
sources:
- name: cluster
  type: kubernetes-resource
  source:
    kubeconfig: |
      # [...]
      users:
      - name: kubernetes-admin
        user:
          exec:
            apiVersion: client.authentication.k8s.io/v1alpha1
            command: aws-iam-authenticator
            args:
              - "token"
              - "-i"
              - "((CLUSTER_ID))"
              - "-r"
              - "((ROLE_ARN))"
```

It is more flexible then specific source params.

- https://github.com/kubernetes-sigs/aws-iam-authenticator#4-set-up-kubectl-to-use-authentication-tokens-provided-by-aws-iam-authenticator-for-kubernetes

Reply:
There seems to be a tension between flexibility and ease of use. I think the parameters make it easier to use this: it removes the need to specify entire kubeconfigs.